Hi Martin,

Le Saturday 04 July 2009 23:37:41 Martin Michlmayr, vous avez écrit :
> I get the following Cobalt build failure with current git:
>
>   CC      arch/mips/cobalt/mtd.o
> cc1: warnings being treated as errors
> In file included from arch/mips/cobalt/mtd.c:22:
> include/linux/mtd/partitions.h:50: warning: ‘struct mtd_info’ declared
> inside parameter list include/linux/mtd/partitions.h:50: warning: its scope
> is only this definition or declaration, which is probably not what you want
> include/linux/mtd/partitions.h:51: warning: ‘struct mtd_info’ declared
> inside parameter list include/linux/mtd/partitions.h:61: warning: ‘struct
> mtd_info’ declared inside parameter list include/linux/mtd/partitions.h:67:
> warning: ‘struct mtd_info’ declared inside parameter list make[1]: ***
> [arch/mips/cobalt/mtd.o] Error 1
> make: *** [arch/mips/cobalt] Error 2
>
> Does anyone know if there's a fix for this already?

I also had that problem and did the following fix, which still applies to
the mtd-2.6 tree, master branch.
--
From: Florian Fainelli <florian@openwrt.org>
Subject: [PATCH] Fix arch/mips/cobalt/mtd.c build failure

This patch fixes a warning in include/linux/mtd/partitions which
results in the following build failure on MIPS:
 CC arch/mips/cobalt/mtd.o
cc1: warnings being treated as errors
In file included from arch/mips/cobalt/mtd.c:22:
include/linux/mtd/partitions.h:50: warning: 'struct mtd_info' declared inside parameter list
include/linux/mtd/partitions.h:50: warning: its scope is only this definition or declaration, which is probably not what you want
include/linux/mtd/partitions.h:51: warning: 'struct mtd_info' declared inside parameter list
include/linux/mtd/partitions.h:61: warning: 'struct mtd_info' declared inside parameter list
include/linux/mtd/partitions.h:67: warning: 'struct mtd_info' declared inside parameter list
make[1]: *** [arch/mips/cobalt/mtd.o] Error 1
make: *** [arch/mips/cobalt] Error 2

Reported-by: Martin Michlmayr <tbm@cyrius.com>
Signed-off-by: Florian Fainelli <florian@openwrt.org>
---
diff --git a/include/linux/mtd/partitions.h b/include/linux/mtd/partitions.h
index af6dcb9..c8eaf44 100644
--- a/include/linux/mtd/partitions.h
+++ b/include/linux/mtd/partitions.h
@@ -10,7 +10,7 @@
 #define MTD_PARTITIONS_H
 
 #include <linux/types.h>
-
+#include <linux/mtd/mtd.h>
 
 /*
  * Partition definition structure:
